WeakAuras/GenericTrigger.lua

```diff
@@ -1272,6 +1302,78 @@ end
 function GenericTrigger.FinishLoadUnload()
 end
 
+do
+  local function ParseCron(pattern)
+    local tests = {}
+    for test in pattern:gmatch("[^ ,]+") do
+      local startString, endString, intervalString = test:match("(%d*)-?(%d*)/?(%d*)")
+      local intervalNumber = tonumber(intervalString) or 1
+      local startNumber = startString == "" and 1 or tonumber(startString) or 1
+      local endNumber = tonumber(endString)
+      if not endNumber then
+        endNumber = intervalString and math.huge or startNumber
```

Comment:
I didn't fully understand what checks you made to decide on whether endNumber should be math.huge or startNumber, maybe I missed something here.

Reply:
An example of test that broke with this version of the function

```Lua
-- test function
local function cron(test, count)
  local startString, endString, intervalString = test:match("(%d*)-?(%d*)/?(%d*)")
  local intervalNumber = tonumber(intervalString) or 1
  local startNumber = startString == "" and 1 or tonumber(startString) or 1
  local endNumber = tonumber(endString)
  if not endNumber then
    endNumber = intervalString and math.huge or startNumber
  end
  if count == 0 then
    print(startString, endString, intervalString)
    print(startNumber, endNumber, intervalNumber)
    print("###")
  end
  return (count >= startNumber and count <= endNumber and (count - startNumber) % intervalNumber == 0)
end
```
```
local test = "5"
for i = 0, 10 do
  print(i, cron(test, i))
end 

5		    -- endString == "" and intervalString == ""
5	inf	1 -- endNumber should be == startNumber
###
0	false
1	false
2	false
3	false
4	false
5	true
6	true
7	true
8	true
9	true
10	true
```

Reply:
Oh right I'll tweak that, the check for "intervalString" is wrong. I need to check that the user actually supplied an interval.

I've pushed a fix. We should squash the commits a bit before merging.